Automattic/jetpack
# Notices: Add many notices generated by Jetpack::state() -- Errors and others.

![screen shot 2016-05-30 at 12 24 27 am](https://cloud.githubusercontent.com/assets/7129409/15639689/f9031750-25fc-11e6-8592-61a007c4035a.png)

`Jetpack::state()` is already in place, and it handles the _one-time_ notices that are triggered by code _outside_ of the app and may be shown for errors or user actions.  It works by setting a cookie that expires immediately after the request is finished.   

This PR takes care of most of the notices handled by Jetpack::state(). It does so by passing over the key generated by the state() method via `window.Initial_State`, and then filtered via switch statements in a component that does not need to be connected to the global state tree.  

It will listen for connection attempt errors and many other things.

**To Test Errors:**
- Disconnect your site.  
- You may force any connection notice by replacing [this fake error](https://github.com/Automattic/jetpack/compare/add/jetpack-state-notices?expand=1#diff-9ba9fa5bb52bd6cf36c6c0bdd5ed4830R3662) with whatever error/message from [this switch](https://github.com/Automattic/jetpack/compare/add/jetpack-state-notices?expand=1#diff-30ae3239dbbcee448073a63cc3fd4bcfR22).  Follow the instructions in those comments to test. 

**To test other notices:** 
- If you "deny" the connection authorization (In the old connection flow only), you will see a notice when you're redirected back.  
- Test the "upgrade" notice by: 
entering `wp jetpack options update version 4.0` in WP CLI.  Then, 
refresh the page.
- Dismissing them should do so.  You should not see them on page refresh or after you navigate away from the page.  

I'm haven't looked exactly how to force an example for the other `state( 'message' )`, but I am confident they will work just the same. 

The icons are still broken when not connected.  Known issue not involved with this PR.  
There will be more to do after this branch.  Like the privacy ones.
